disk/tests/test_filter.py

```diff
@@ -53,12 +53,12 @@ def test_ignore_empty_regex():
     }
     c = Disk('disk', None, {}, [instance])
 
-    assert c._file_system_whitelist.pattern == 'test'
-    assert c._file_system_blacklist.pattern == 'test|iso9660$'
-    assert c._device_whitelist.pattern == 'test'
-    assert c._device_blacklist.pattern == 'test'
-    assert c._mount_point_whitelist.pattern == 'test'
-    assert c._mount_point_blacklist.pattern == 'test'
+    assert c._file_system_whitelist == re.compile('test', re.I)
+    assert c._file_system_blacklist == re.compile('test|iso9660$', re.I)
+    assert c._device_whitelist == re.compile('test', IGNORE_CASE)
+    assert c._device_blacklist == re.compile('test', IGNORE_CASE)
+    assert c._mount_point_whitelist == re.compile('test', IGNORE_CASE)
+    assert c._mount_point_blacklist == re.compile('test', IGNORE_CASE)
```

Comment:
Comparing compiled regex in python seems not reliable.

```python
import re

abc_re1 = re.compile('abc')
abc_re2 = re.compile('abc')
assert abc_re1 == abc_re2  # OK
for i in range(10000): re.compile(str(i))
abc_re3 = re.compile('abc')

assert abc_re1 == abc_re3  # FAIL

# maybe something like this could help
def assert_pattern(regex_a, regex_b):
    assert (regex_a.pattern, regex_a.flags) == (regex_b.pattern, regex_b.flags)

assert_pattern(abc_re1, abc_re2) # OK
```

Reply:
I'll change it, but this works correctly in Python 3 😉